core/src/test/java/org/elasticsearch/index/query/MissingQueryBuilderTest.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.Query;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.is;
+
+public class MissingQueryBuilderTest extends BaseQueryTestCase<MissingQueryBuilder> {
+
+    @Override
+    protected MissingQueryBuilder doCreateTestQueryBuilder() {
+        MissingQueryBuilder query  = new MissingQueryBuilder(getRandomFieldName());
+        if (randomBoolean()) {
+            query.nullValue(randomBoolean());
+        }
+        if (randomBoolean()) {
+            query.existence(randomBoolean());
+        }
+        // cannot set both to false
+        if ((query.nullValue() == false) && (query.existence() == false)) {
+            query.existence(!query.existence());
+        }
+        return query;
+    }
+
+    private String getRandomFieldName() {
+        switch (randomIntBetween(0, 3)) {
+            case 0:
+                return BOOLEAN_FIELD_NAME;
+            case 1:
+                return STRING_FIELD_NAME;
+            case 2:
+                return INT_FIELD_NAME;
+            case 3:
+                return DOUBLE_FIELD_NAME;
+        }
+        return randomAsciiOfLengthBetween(1, 10);
+    }
+
+    @Override
+    protected Query doCreateExpectedQuery(MissingQueryBuilder queryBuilder, QueryParseContext context) throws IOException {
+        return MissingQueryBuilder.newFilter(context, queryBuilder.fieldPattern(), queryBuilder.existence(), queryBuilder.nullValue());
```

Comment:
I think it is more of a general discussion around this issue. @cbuescher does raise some valid points here, but are we settle as to what we want to do? If you ask me I'm not very satisfied with either option.